include/GIL/Instructions/CondBrInst.hpp

```diff
@@ -11,36 +12,178 @@ namespace glu::gil {
+
+        return new (mem) CondBrInst(condition, thenBlock, elseBlock, {}, {});
+    }
+
+    /// @brief Static factory method to create a CondBrInst with arguments for
+    /// both branches.
+    ///
+    /// @param arena The memory arena to allocate from.
+    /// @param condition The condition value that determines which branch to
+    /// take.
+    /// @param thenBlock The basic block to branch to if the condition is true.
+    /// @param elseBlock The basic block to branch to if the condition is false.
+    /// @param thenArgs The arguments to pass to the then block.
+    /// @param elseArgs The arguments to pass to the else block.
+    static CondBrInst *create(
+        llvm::BumpPtrAllocator &arena, Value condition, BasicBlock *thenBlock,
+        BasicBlock *elseBlock, llvm::ArrayRef<Value> thenArgs,
+        llvm::ArrayRef<Value> elseArgs
+    )
     {
+        auto totalSize
+            = totalSizeToAlloc<Value, Value>(thenArgs.size(), elseArgs.size());
+        void *mem = arena.Allocate(totalSize, alignof(CondBrInst));
+
+        return new (mem)
+            CondBrInst(condition, thenBlock, elseBlock, thenArgs, elseArgs);
+    }
+
+    // Helper methods to access the trailing objects
+    Value *getThenArgsPtr() { return getTrailingObjects<Value>(); }
+    Value const *getThenArgsPtr() const { return getTrailingObjects<Value>(); }
+
+    Value *getElseArgsPtr()
+    {
+        return getTrailingObjects<Value>() + _thenArgsCount;
+    }
+    Value const *getElseArgsPtr() const
+    {
+        return getTrailingObjects<Value>() + _thenArgsCount;
+    }
+
+    Value getCondition() const { return condition; }
+    BasicBlock *getThenBlock() const { return thenBlock; }
+    BasicBlock *getElseBlock() const { return elseBlock; }
+
+    llvm::ArrayRef<Value> getThenArgs() const
+    {
+        return llvm::ArrayRef<Value>(getThenArgsPtr(), _thenArgsCount);
+    }
+
+    llvm::ArrayRef<Value> getElseArgs() const
+    {
+        return llvm::ArrayRef<Value>(getElseArgsPtr(), _elseArgsCount);
+    }
+
+    bool hasBranchArgs() const
+    {
+        return _thenArgsCount > 0 || _elseArgsCount > 0;
     }
 
     static bool classof(InstBase const *inst)
     {
         return inst->getKind() == InstKind::CondBrInstKind;
     }
 
-    size_t getOperandCount() const override { return 3; }
-    Operand getOperand([[maybe_unused]] size_t index) const override
+    size_t getOperandCount() const override
+    {
+        // 3 base operands (condition, thenBlock, elseBlock) + branch arguments
+        return 3 + _thenArgsCount + _elseArgsCount;
+    }
+
+    Operand getOperand(size_t index) const override
     {
         if (index == 0)
             return condition;
         if (index == 1)
             return thenBlock;
         if (index == 2)
             return elseBlock;
-        else
-            assert(false && "Invalid operand index");
+
+        // Handle then arguments
+        if (index < 3 + _thenArgsCount) {
```

Comment:
```suggestion
        if (index - 3 < _thenArgsCount) {
```
Consistent with what’s below and mathematically equivalent